src/base/random.test.cc

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2024 CNRS
+ * All rights reserved
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met: redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer;
+ * redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution;
+ * neither the name of the copyright holders nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <gtest/gtest-spi.h>
+#include <gtest/gtest.h>
+
+#include <unordered_map>
+
+#include "base/gtest/logging.hh"
+#include "base/random.hh"
+
+using namespace gem5;
+
```

Comment:
Could you add equivalent tests for floating points?

Reply:
FP API already provides numbers between 0.0 and 1.0, but currently the API does not let the user set another range. I think I did not get what exactly you wanted to test.

Reply:
I meant adding something like the following:
```
TEST(RandomCtor, DoubleDefaultConstruct)
{
    // Init seed was default seed 5489
    Random dut;
    // First number corresponds to the default seed
    ASSERT_EQ(dut.random<double>(), 0,837237944); // 14514284786278117030llu / max<uint64_t>
}

TEST(RandomCtor, FloatDefaultConstruct)
{
    // Init seed was default seed 5489
    Random dut;
    // First number corresponds to the default seed
    ASSERT_EQ(dut.random<float>(), the_expected_float_value);
}
```

Reply:
I added the tests, but they seem redundant to me : since float/double are currently based on integer random number generation, the fact that the first integer is the expected one should imply the first float and double are the expected one. Maybe I'm missing out the reason for adding this test : is this about ensuring FP behavior is consistent across different potential host ISA for random number generation ?

Reply:
The way that I see it, we can´t guarantee that the current behavior will be kept forever, so the test helps make sure that if it is ever changed (for example, to provide uniformity in the distribution), it is done consciously, not carelessly.